consensus/service/src/byzantine_ledger.rs

```diff
@@ -520,8 +520,13 @@ impl<
 
                 // Clear any pending values that might no longer be valid.
                 let tx_manager = self.tx_manager.clone();
-                self.pending_values
-                    .retain(|tx_hash| tx_manager.validate_tx_by_hash(tx_hash).is_ok());
+                self.pending_values.retain(|tx_hash| {
```

Comment:
This might perform better if the lock is moved outside of the loop.

Reply:
Nice. Will do.